docs/db2graph/db2graph.rst

```diff
@@ -0,0 +1,279 @@
+Db2Graph: Database to Graph conversion tool
+=========================
+
+Introduction
+""""""""""""""""""""
+
+**Db2Graph** converts **relational databases** into **graphs as sets of triples** which can be used as **input datasets for Marius**, allowing streamlined preprocessing from database to Marius. Db2Graph comes with Marius but can be used as a standalone tool. Conversion with Db2Graph is achieved in the following steps: 
+
+#. Users import/create the database locally
+
+#. Users define the configuration file and entity/edge SQL SELECT queries
+
+#. Db2Graph executes the SQL SELECT queries
+
+#. Db2Graph transforms the result set of queries into sets of triples
+
+Below we lay out the requirements, definitions, and steps for using Db2Graph, and a real example use case:
+
+Requirements
+""""""""""""""""""""
+
+Db2Graph currently supports graph conversion from three relational database management systems: **MySQL**, **MariaDB**, and **PostgreSQL**. Db2Graph requires no installation but additional packages not included in the Marius installation. The following packages are required:
+
+* python >= 3.6  (included in Marius)
+* pandas  (included in Marius)
+* hydra  (included in Marius)
+* psutil  (included in Marius)
+* MySQL connector >= 8.0
+* Psycopg2 >= 2.9
+
+System Design
+""""""""""""""""""""
+
+Db2Graph classifies a graph into the following four types:
+
+* Entity Nodes: Nodes that are globally unique either by UUID's (Universally Unique Identifier) or in the format of ``table-name_col-name_val``. In a graph, entity nodes either point to other entity nodes or feature values, or are pointed to by other entity nodes.
+* Feature Values: Features values define the features of entities nodes and are not globally unique. Feature Values can only be pointed to by entity nodes.
```

Comment:
What exactly are these feature values? Can you give an example here in the docs?

I see in the tests they are these attributes e.g. `orders_item_chicken_breast -> costs -> 12`. 

I'm not so sure that encoding these attributes as a triples is the best way to represent them. As this means we have nodes in the graph for all numerical values which appear in the database (a node for 0, 1, 2, ...). I'm not sure this makes sense. @thodrek what are your thoughts on how to best handle this?

Reply:
Removing this type of edges is the best solution at this stage.

Reply:
Removed and new changed pushed